MiniApp/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -75,6 +76,82 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
+                case .failure(let error):
+                    self.executeJavaScriptCallback(responseStatus: .onError, messageId: callBackId, response: error.localizedDescription)
+                }
+            })
+        } else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callBackId, response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
+        }
+    }
+
+    private func sendMessageToContactId(with callBackId: String, parameters: RequestParameters?) {
+        if let message = parameters?.messageToContact {
+            if let contactId = parameters?.contactId {
+                hostAppMessageDelegate?.sendMessageToContact(contactId, message: message) { result in
+                    switch result {
+                    case .success(let contact):
+                        guard let data = try? JSONEncoder().encode(contact),
+                              let response = String(data: data, encoding: .utf8) else {
+                            return self.executeJavaScriptCallback(
+                                    responseStatus: .onError,
+                                    messageId: callBackId,
+                                    response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
+                        }
+                        self.executeJavaScriptCallback(responseStatus: .onSuccess, messageId: callBackId, response: response)
+                    case .failure(let error):
+                        self.executeJavaScriptCallback(responseStatus: .onError, messageId: callBackId, response: error.localizedDescription)
+                    }
+                }
+            } else {
+                executeJavaScriptCallback(responseStatus: .onError, messageId: callBackId, response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
+            }
+        } else {
+            executeJavaScriptCallback(responseStatus: .onError, messageId: callBackId, response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
+        }
+    }
+
+    private func sendMessageToMultipleContacts(with callBackId: String, parameters: RequestParameters?) {
+        if let message = parameters?.messageToContact {
+            hostAppMessageDelegate?.sendMessageToMultipleContacts(message, completionHandler: { result in
+                switch result {
+                case .success(let contactIds):
+                    guard let data = try? JSONEncoder().encode(contactIds),
+                          let response = String(data: data, encoding: .utf8) else {
+                        return self.executeJavaScriptCallback(
+                                responseStatus: .onError,
+                                messageId: callBackId,
+                                response: getMiniAppErrorMessage(MiniAppJavaScriptError.unexpectedMessageFormat))
```

Comment:
Do you think its okay to return `unexpectedMessageFormat` even if user cancelled the operation ?

Reply:
oh, if the user cancelled it will return nil, and `JSONEncoder` will encode it to `null` ;)